packages/block-editor/src/hooks/align.js

```diff
@@ -207,6 +212,130 @@ export const withDataAlign = createHigherOrderComponent(
 		return <BlockListBlock { ...props } wrapperProps={ wrapperProps } />;
 	}
 );
+export function AlignmentVisualizer( {
+	value = 'none',
+	allowedAlignments,
+	clientId,
+} ) {
+	const layout = useLayout();
+	const blockName = useSelect(
+		( select ) => select( blockEditorStore ).getBlockName( clientId ),
+		[ clientId ]
+	);
+
+	const [ editorWidth, setEditorWidth ] = useState( 0 );
+	const ref = useRefEffect( ( node ) => {
+		const { ownerDocument } = node;
+		const { defaultView } = ownerDocument;
+		const editorWrapperElement = ownerDocument.querySelector(
+			'.editor-styles-wrapper'
+		);
+		const updateWidth = () =>
+			setEditorWidth( editorWrapperElement.offsetWidth );
+		updateWidth();
+
+		const resizeObserver = defaultView.ResizeObserver
+			? new defaultView.ResizeObserver( updateWidth )
+			: undefined;
+		resizeObserver?.observe( editorWrapperElement );
+
+		return () => {
+			resizeObserver?.disconnect();
+		};
+	}, [] );
+
+	const blockAllowedAlignments = getValidAlignments(
+		getBlockSupport( blockName, 'align' ),
+		hasBlockSupport( blockName, 'alignWide', true )
+	);
+
+	// Allow override of `blockAllowedAlignments`. The image block doesn't use
+	// alignment block supports, so this allows the image block to directly
+	// declare what it supports.
+	const availableAlignments = useAvailableAlignments(
+		allowedAlignments ?? blockAllowedAlignments
+	);
+
+	const alignments = useMemo( () => {
+		return [ 'none', 'wide', 'full' ]
+			.map( ( name ) => {
+				if ( name === 'none' ) {
+					return {
+						name,
+						label: __( 'Content width' ),
+						style: {
+							width: layout.contentSize,
+						},
+					};
+				}
+				if ( name === 'wide' ) {
+					return {
+						name,
+						label: __( 'Wide width' ),
+						style: {
+							width: layout.wideSize,
```

Comment:
It seems a shame that the theme.json properties are `contentSize` and `wideSize`, while the alignment control uses `none` and `wide`. There's a lot of manual stitching together required to make it work as expected. [Custom alignments](https://github.com/WordPress/gutenberg/issues/38912) would also be hard to support.

Another note is that some of this mapping could be the responsibility of `useAvailableAlignments`